doc/ssh_config.md

```diff
@@ -0,0 +1,230 @@
+			- [1.13.2. <a name='SystemTestcases'></a>System Test cases](#1132-system-test-cases)
+		- [1.14. <a name='OpenActionitems-ifany'></a>Open/Action items - if any](#114-openaction-items---if-any)
+###  1.1. <a name='Revision'></a>Revision  
+
+###  1.2. <a name='Scope'></a>Scope  
+
+This hld doc for ssh server global configurations describes the requirements, architecture and general flow details of ssh global config in SONIC OS based switches.
+
+###  1.3. <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations 
+
+	SSH - secure shell
+	TCP - Transmission Control protocol	
+	
+###  1.4. <a name='Overview'></a>Overview 
+
+We want to allow configuring ssh server global settings. This will feature will include 5 configurations, but can be extended easily to include additional configurations.
+
+###  1.5. <a name='Requirements'></a>Requirements
+
+This feature requires a dedicated table in the configuration DB, and enhancements of hostcfg demon, in order to allow modifing the relvant ssh configuration files. In order to override ssh configurations, we need to have write access to ssh config files such as /etc/ssh/sshd_config
+
+###  1.6. <a name='ArchitectureDesign'></a>Architecture Design 
+####  1.6.1. <a name='ConfigModules'></a>Configuration modules
+![ssh_config_flow](ssh_config_flow.png)
+
+We want to enhance configDB to include table for ssh server global configurations. In addition, hostcfg demon will include a dedicated flow in order to modify ssh config files, once ssh server global config table entries are changed.
+
+###  1.7. <a name='High-LevelDesign'></a>High-Level Design 
+
+We want to enable global ssh server configuration in SONIC. In order to do so will touch few areas in the system:
+1. configDB - to include a dedicated table for configurations
+2. hostcfg demon - to update ssh config files once configDB relevant areas are modified (and for this feature, ssh global config table)
+3. OS ssh config files - specific for this stage we are only /etc/ssh/sshd_config is going to be modifed by the hostcfg demon.
+4. OS ssh service - to be restarted after each configuration change.
+
+Note:
+
+The Daemon is running in the host (without container) that matches this feature, because it is basically writing policies on ssh config files from the OS.
+##### Flow diagram
+![ssh_flow_community](ssh_flow_community.png)
+#### 1.7.1 <a name='Flow description'></a>Flow description
+When the feature is enabled, by modifying the DB manually, user will set ssh server policies/configuration (see options below) by modifing CONF_DB in SSH_GLOBAL_TABLE.
+
+The hostcfgd daemon will be extended to listen to ssh policies/configurations from SSH_GLOBAL table, parse the inputs and set the new policies to ssh config files, and update ssh server afterwards.
+
+
+#### 1.7.2 <a name='SSH global policies'></a>SSH global policies
+
+We want to enable configuring the following policies, with default values are taken from OS (Debian):

```

Comment:
Can you please mention if the current SONiC defaults match with the defaults that you are proposing. This will highlight that new design is backward compatible.

Reply:
Current design should match debian standards, if these are in conflict - do we still want to take SONiC basic defaults?